hbase-common/src/main/java/org/apache/hadoop/hbase/io/TagCompressionContext.java

```diff
@@ -81,7 +81,8 @@ public void compressTags(OutputStream out, byte[] in, int offset, int length) th
   public void compressTags(OutputStream out, ByteBuffer in, int offset, int length)
     throws IOException {
     if (in.hasArray()) {
-      compressTags(out, in.array(), offset, length);
+      // Offset we are given is relative to ByteBuffer#arrayOffset
+      compressTags(out, in.array(), in.arrayOffset() + offset, length);
```

Comment:
This is one of several instances where we don't consider `arrayOffset` when accessing the contents of a byte[] backing a ByteBuffer. However I don't think they are real bugs at this time because we are not `slice()`ing buffers in way where offsets would be nonzero, not yet.